### DISCUSSION TITLE
revalidatePath doesnt invalidates full-route cache

### DISCUSSION THREAD

**Discussion by dr-azbest** [vercel/next.js#69595] [Help]
### Summary

Hi guys, I'm trying to understand what happend here and if it is bug or not. Lets say I have a list of tasks fetched on server component and passed to client side component . Let it be /tasks/1 route and GET /api/tasks/1. Whenever i hit delete or add task action on client side i make appropriate api call and revalidatePath("/tasks/1). Now, when I go to this /tasks/1 route again I can log that data was indeed invalidated and it is fetched again, and correct data state is indeed passed to client side component. The problem is I got hydration error because list content changed and RSC Payload do no match. How i understand docs, what happend here is, nextjs creates full route cache with RSC Payload at first load, and at second route load before fetching revalidated data it serves RSC Payload from previous generation. It is also visible on client side I got old state served at first paint. Shouldnt revalidatePath invalidate RSC Payload as well ?

**Comment by suprunchuk** [comment]
This behavior is by design for performance reasons. Next.js aims to serve content as quickly as possible, so it uses the cached version first while fetching updated data in the background.

To address this issue, you have a few options:

1. **Use [useEffect](https://react.dev/reference/react/useEffect) to handle client-side updates:**
```js
'use client';

import { useEffect, useState } from 'react';

export default function TaskList({ initialTasks }) {
  const [tasks, setTasks] = useState(initialTasks);

  useEffect(() => {
    // Fetch updated tasks
    fetch('/api/tasks/1')
      .then(response => response.json())
      .then(data => setTasks(data));
  }, []);

  // Render tasks
}
```

**or**

2. **Use Next.js's [unstable_noStore](https://nextjs.org/docs/app/api-reference/functions/unstable_noStore) to disable caching for this route:**

```js
import { unstable_noStore as noStore } from 'next/cache';

export default async function TaskList() {
  noStore();
  const tasks = await fetch('/api/tasks/1').then(res => res.json());
  
  // Render tasks
}
```

**Comment by dr-azbest** [comment]
Is there any way I can invalidate this cached RSC payload manualy? Above options dosent satisfy me as I want this whole page to be cached. I think docs rather suggest RSC payload should be invalidated:
![image](https://github.com/user-attachments/assets/40798a4e-41ca-4be3-a5a4-76dc0aa7ab71)

Above we can see that RSC payload is part of full-route-chache so below invalidation should also IMO invalidate it.
> [Invalidation](https://nextjs.org/docs/app/building-your-application/caching#invalidation)
There are two ways you can invalidate the Full Route Cache:

>[Revalidating Data](https://nextjs.org/docs/app/building-your-application/caching#revalidating): Revalidating the [Data Cache](https://nextjs.org/docs/app/building-your-application/caching#data-cache), will in turn invalidate the Router Cache by re-rendering components on the server and caching the new render output.
Redeploying: Unlike the Data Cache, which persists across deployments, the Full Route Cache is cleared on new deployments.

Isnt it obvious if I manualy invalidate full-route cache, I will probably got new data and rsc payload will mismatch?
